app/code/Magento/Catalog/Console/Command/ImagesResizeCommand.php

```diff
@@ -18,31 +22,31 @@ class ImagesResizeCommand extends \Symfony\Component\Console\Command\Command
     protected $productCollectionFactory;
 
     /**
-     * @var \Magento\Catalog\Api\ProductRepositoryInterface
+     * @var QueueFactory
      */
-    protected $productRepository;
+    protected $queueFactory;
 
     /**
-     * @var \Magento\Catalog\Model\Product\Image\CacheFactory
+     * @var Options
      */
-    protected $imageCacheFactory;
+    protected $options;
 
     /**
      * @param \Magento\Framework\App\State $appState
      * @param \Magento\Catalog\Model\ResourceModel\Product\CollectionFactory $productCollectionFactory
-     * @param \Magento\Catalog\Api\ProductRepositoryInterface $productRepository
-     * @param \Magento\Catalog\Model\Product\Image\CacheFactory $imageCacheFactory
+     * @param QueueFactory $queueFactory
+     * @param Options $options
      */
     public function __construct(
         \Magento\Framework\App\State $appState,
         \Magento\Catalog\Model\ResourceModel\Product\CollectionFactory $productCollectionFactory,
-        \Magento\Catalog\Api\ProductRepositoryInterface $productRepository,
```

Comment:
To maximize backwards compatibility removing non-optional constructor parameter is not allowed.
If this is not needed anymore please remove the type hint and mark it as a deprecated.